release/ray_release/command_runner/_anyscale_job_wrapper.py

```diff
@@ -0,0 +1,348 @@
+    try:
+        subprocess.run(
+            [
+                "python",
+                "prometheus_metrics.py",
+                str(time_taken),
+                "--path",
+                os.environ["METRICS_OUTPUT_JSON"],
+            ],
+            timeout=metrics_timeout,
+            check=True,
+        )
+        return True
+    except subprocess.SubprocessError:
+        logger.exception("Couldn't collect metrics.")
+        return False
+
+
+# Has to be here so it can be pickled
+def _run_bash_command_subprocess(command: str, timeout: float):
+    try:
+        subprocess.run(command, check=True, timeout=timeout)
+        return_code = 0
+    except subprocess.TimeoutExpired:
+        return_code = TIMEOUT_RETURN_CODE
+    except subprocess.CalledProcessError as e:
+        return_code = e.returncode
+    print(f"Subprocess return code: {return_code}", file=sys.stderr)
+    sys.exit(return_code)
+
+
+def run_bash_command(workload: str, timeout: float):
+    timeout = timeout if timeout > 0 else None
+    cwd = Path.cwd()
+    workload_path = cwd / "workload.sh"
+    workload_path = workload_path.resolve()
+    with open(workload_path, "w") as fp:
+        fp.write(workload)
+
+    command = ["bash", "-x", str(workload_path)]
+    logger.info(f"Running command {workload}")
+
+    # Pop job's runtime env to allow workload's runtime env to take precedence
+    # TODO: Confirm this is safe
+    os.environ.pop("RAY_JOB_CONFIG_JSON_ENV_VAR", None)
+
+    # We use multiprocessing with 'spawn' context to avoid
+    # forking (as happens when using subprocess directly),
+    # because that messes up Ray interactions and causes
+    # deadlocks.
+    return_code = None
+    try:
+        ctx = multiprocessing.get_context("spawn")
+        p = ctx.Process(target=_run_bash_command_subprocess, args=(command, timeout))
+        p.start()
+        logger.info(f"Starting process {p.pid}.")
+        # Add a little extra to the timeout as _run_bash_command_subprocess
+        # also has a timeout internally and it's cleaner to use that
+        p.join(timeout=timeout + 10)
+    except multiprocessing.TimeoutError:
+        return_code = TIMEOUT_RETURN_CODE
+    except multiprocessing.ProcessError:
+        pass
+    finally:
+        if p.is_alive():
+            logger.warning(f"Terminating process {p.pid} forcefully.")
+            p.terminate()
+        if return_code is None:
+            return_code = p.exitcode
+        os.remove(str(workload_path))
+    logger.info(f"Process {p.pid} exited with return code {return_code}.")
+    assert return_code is not None
+    return return_code
+
+
+def run_prepare_commands(
+    prepare_commands: List[str], prepare_commands_timeouts: List[float]
+) -> Tuple[bool, List[int], float]:
+    prepare_return_codes = []
+    prepare_passed = True
+    prepare_time_taken = None
+    if prepare_commands:
+        logger.info("### Starting prepare commands ###")
+        for prepare_command, timeout in zip(
```

Comment:
```suggestion
    if not prepare_commands:
        return prepare_passed, prepare_return_codes, prepare_time_taken
        
    logger.info("### Starting prepare commands ###")
    for prepare_command, timeout in zip(
    # ...
```
Let's remove some nesting here